doc/man3/EVP_DigestInit.pod

```diff
@@ -72,120 +53,172 @@ EVP_get_digestbyobj - EVP digest routines
-are no longer linked this function is only retained for compatibility
-reasons.
+Returns the B<EVP_MD> structure corresponding to the passed B<EVP_MD_CTX>.
 
-EVP_md2(), EVP_md5(), EVP_sha1(), EVP_sha224(), EVP_sha256(),
-EVP_sha384(), EVP_sha512(), EVP_sha3_224(), EVP_sha3_256(),
-EVP_sha3_384(), EVP_sha3_512(), EVP_mdc2(), EVP_ripemd160(),
-EVP_blake2b512(), and EVP_blake2s256() return B<EVP_MD> structures for
-the MD2, MD5, SHA1, SHA224, SHA256, SHA384, SHA512, SHA3-224, SHA3-256,
-SHA3-384, SHA3-512, MDC2, RIPEMD160, BLAKE2b-512, and BLAKE2s-256 digest
-algorithms respectively.
+=item EVP_MD_pkey_type()
 
-EVP_md_null() is a "null" message digest that does nothing: i.e. the hash it
-returns is of zero length.
+Returns the NID of the public key signing algorithm associated with this
+digest. For example EVP_sha1() is associated with RSA so this will return
+B<NID_sha1WithRSAEncryption>. Since digests and signature algorithms are no
+longer linked this function is only retained for compatibility reasons.
 
-EVP_get_digestbyname(), EVP_get_digestbynid() and EVP_get_digestbyobj()
-return an B<EVP_MD> structure when passed a digest name, a digest NID or
-an ASN1_OBJECT structure respectively.
+=item EVP_md_null()
+
+A "null" message digest that does nothing: i.e. the hash it returns is of zero
+length.
+
+=item EVP_get_digestbyname(),
+EVP_get_digestbynid(),
+EVP_get_digestbyobj()
+
+Returns an B<EVP_MD> structure when passed a digest name, a digest B<NID> or an
+B<ASN1_OBJECT> structure respectively.
+
+=back
 
 =head1 RETURN VALUES
 
-EVP_DigestInit_ex(), EVP_DigestUpdate() and EVP_DigestFinal_ex() return 1 for
+=over 4
+
+=item EVP_DigestInit_ex(),
+EVP_DigestUpdate(),
+EVP_DigestFinal_ex()
+
+Returns 1 for
 success and 0 for failure.
 
-EVP_MD_CTX_ctrl() returns 1 if successful or 0 for failure.
+=item EVP_MD_CTX_ctrl()
+
+Returns 1 if successful or 0 for failure.
+
+=item EVP_MD_CTX_copy_ex()
 
-EVP_MD_CTX_copy_ex() returns 1 if successful or 0 for failure.
+Returns 1 if successful or 0 for failure.
 
-EVP_MD_type(), EVP_MD_pkey_type() and EVP_MD_type() return the NID of the
-corresponding OBJECT IDENTIFIER or NID_undef if none exists.
+=item EVP_MD_type(),
+EVP_MD_pkey_type(),
+EVP_MD_type()
 
-EVP_MD_size(), EVP_MD_block_size(), EVP_MD_CTX_size() and
-EVP_MD_CTX_block_size() return the digest or block size in bytes.
+Returns the NID of the corresponding OBJECT IDENTIFIER or NID_undef if none
+exists.
 
-EVP_md_null(), EVP_md2(), EVP_md5(), EVP_sha1(),
-EVP_mdc2(), EVP_ripemd160(), EVP_blake2b512(), and EVP_blake2s256() return
-pointers to the corresponding EVP_MD structures.
+=item EVP_MD_size(),
+EVP_MD_block_size(),
+EVP_MD_CTX_size(),
+EVP_MD_CTX_block_size()
 
-EVP_get_digestbyname(), EVP_get_digestbynid() and EVP_get_digestbyobj()
-return either an B<EVP_MD> structure or NULL if an error occurs.
+Returns the digest or block size in bytes.
+
+=item EVP_md_null()
+
+Returns a pointer to the corresponding B<EVP_MD> structures.
```

Comment:
I don't understand this sentence, is it in the wrong context.

Reply:
@bernd-edlinger I think there's something wrong with the diff?

I'm seeing the right description here: https://github.com/openssl/openssl/pull/4628/files#diff-b285e033e98ee2d9eff6aec1a7bd6dbfR156

Reply:
Yes, the description looks right, but under return values I see something like:

```
EVP_md_null()
    Returns a pointer to the corresponding EVP_MD structures
```

That sounds a bit funny I think you mean to say something about the return value of any
EVP_<digest> function here?

Reply:
@bernd-edlinger in the diff the content is:
"A "null" message digest that does nothing: i.e. the hash it returns is of zero length."

Reply:
No, I mean the return value section,
previously the text was:

```
EVP_md_null(), EVP_md2(), EVP_md5(), EVP_sha1(),
EVP_mdc2(), EVP_ripemd160(), EVP_blake2b512(), and EVP_blake2s256() return
pointers to the corresponding EVP_MD structures.
```

which made sense, but now it is:

```
EVP_md_null()
Returns a pointer to the corresponding B<EVP_MD> structures.
```

So, unless I am completely mistaken,
the return value is the EVP_MD structure of the NULL MD or something.

Reply:
Sorry I was entirely confused. I've now fixed hopefully the right issue 😉